@xen-orchestra/lite/src/App/TreeView.tsx

```diff
@@ -109,6 +122,8 @@ const TreeView = withState<State, Props, Effects, Computed, ParentState, ParentE
       },
       hostsByPool: state => state.objectsByType?.get('host')?.groupBy(host => host.$pool.$id),
       pools: state => state.objectsByType?.get('pool'),
+      selectedNodes: ({ _selectedNodes }, { defaultSelectedNodes }) =>
+        _selectedNodes ?? (defaultSelectedNodes === undefined ? [] : defaultSelectedNodes),
```

Comment:
Why not `_selectedNodes ?? defaultSelectedNodes`?

Reply:
I suggest `defaultSelectedNodes === undefined ? [] : _selectedNodes ?? defaultSelectedNodes` instead ?

Because with the actual condition, it effectively can't work.
 in case a node is selected, we will always use it to determine the highlighted node. Then even if we go to `infrastructure` we will highlight the latest selected VM.

Reply:
I don't understand what you mean, can you give an example?

Reply:
in case where `_selectedNodes ?? defaultSelectedNodes`
If we select a VM, then `_selectedNodes` will no longer be `undefined`.
If we click to the `infrastructure` link, `_selectedNodes` won't be `undefined`, so we will still highlight the last VM selected.
That why i think it's important to check `defaultSelectedNodes` first. 
In this case `defaultSelectedNodes` will be undefined, because he is based on the URL